### PR TITLE
Ensure we don't index noindex nodes

### PR DESCRIPTION
**Pull request by matthew-hallsworth**
This PR hooks into the metatag information stored about the node. If the node has it's own metatag information stored and the 'noindex' value is checked, then we should not index the node for the xmlsitemap.

It's not a tight dependency on the metatag module because it's doing a node load and checking for the presence of the node values against the node object prior to making any assumptions.

@ali-sarfaraz @aaron-meeking please review

Note that this is part of composer installing the xmlsitemap lib ourselves from our own repo, since the 7.x branch of contrib is abandonware now for Drupal 7.

### CODE REVIEW

**Comment by matthew-hallsworth** [comment]
For @rubylamadora:

This is working and ready to go, what's left to do is:
- get it reviewed!
- merge this into 7.x-2.x
- make a new release - which is '7.x-2.7' and '2.7' for composer
- update the version in the endurance branch being installed
- test in endurance branch

File: xmlsitemap_node/xmlsitemap_node.module [hunk 1]
```diff
@@ -73,8 +73,14 @@ function xmlsitemap_node_xmlsitemap_process_node_links(array $nids) {
     foreach ($nids_chunks as $chunk) {
       $nodes = node_load_multiple($chunk);
       foreach ($nodes as $node) {
-        $link = xmlsitemap_node_create_link($node);
-        xmlsitemap_link_save($link, array($link['type'] => $node));
+        $process = true;
+        if ($node->metatags != [] && isset($node->metatags['en']['robots']) && $node->metatags['en']['robots']['value']['noindex'] == 1) {
```
Comment (rubylamadora) [inline xmlsitemap_node/xmlsitemap_node.module]: Will have to test this on ATA as based on my testing with creating a new page, having "en" as default language is not always the case. Will look into bypassing the language part in here Or as what @ali-sarfaraz suggested look into other ways to load the language dynamically.

File: xmlsitemap_node/xmlsitemap_node.module [hunk 2]
```diff
@@ -73,6 +73,13 @@ function xmlsitemap_node_xmlsitemap_process_node_links(array $nids) {
     foreach ($nids_chunks as $chunk) {
       $nodes = node_load_multiple($chunk);
       foreach ($nodes as $node) {
+        $lang = isset($node->language) ? $node->language : LANGUAGE_NONE;
```
Comment (rubylamadora) [inline xmlsitemap_node/xmlsitemap_node.module]: $lang will dynamically get "en" or "und" if undefined.

**Comment by rubylamadora** [comment]
Hey @ali-sarfaraz, can you have another look with the updated changes please.

**Comment by rubylamadora** [comment]
Hey @matthew-hallsworth, I need you opinion on this regarding how to unit test our custom functions. As this is already separated from Drupal, xmlsitemap has its own tests which require DrupalTestHelper which is within endurance at the moment. Do we need to pull down DrupalTest module (which I haven't look into yet) from here or do you feel like creating our own unit test instead?

**Comment by matthew-hallsworth** [comment]
@rubylamadora I'm not sure, normally I would propose using phpunit and writing some mocks for the node load, since that would be in our knowledge to do, but if DrupalTestModule is easy to implement maybe that is the way to do it?

What does @ali-sarfaraz @aaron-meeking think?

**Comment by rubylamadora** [comment]
> @rubylamadora I'm not sure, normally I would propose using phpunit and writing some mocks for the node load, since that would be in our knowledge to do, but if DrupalTestModule is easy to implement maybe that is the way to do it?
> 
> What does @ali-sarfaraz @aaron-meeking think?

I tried adding Drupal test module and it looks like it's not straightforward to setup. I'll go with adding our own unit test as we go. This makes sense to me as all the functionality existing in xmlsitemap should be well tested before they release :P Will add phpunit and test the custom validation.

File: tests/xmlsitemap_node/XmlsitemapNodeTest.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * @file
+ * Unit tests for the xmlsitemap_node module.
+ */
+
+namespace Tests\xmlsitemap_node;
+
+include './xmlsitemap_node/xmlsitemap_node.module';
```
Comment (rubylamadora) [inline tests/xmlsitemap_node/XmlsitemapNodeTest.php]: adding file as xmlsitemap don't have classes.

**Comment by aaron-meeking** [comment]
@rubylamadora - looks pretty straight forward to me. Nice work. I'm happy with those changes.

**Pull request merged**